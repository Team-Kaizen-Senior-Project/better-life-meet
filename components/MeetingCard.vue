<script setup lang="ts">
	import type { Meeting } from '~/types'
	const { display: displayDate, dayjs } = useDate()
	const { deleteMeeting: deleteMeetingApi } = useApi()
	const { state: customerState, isLeader } = usePodStore()
	const pod = customerState.pod!

	const deleting = ref<boolean>(false)

	const props = defineProps<{
		meeting?: Meeting
	}>()

	const emit = defineEmits<{
		refresh: []
	}>()

	const video = useVideoStore()

	const isLive = computed(() => {
		if (!props.meeting) return false
		const start = props.meeting.startTime
		const end = props.meeting.endTime
		return dayjs().isBetween(start.isoString, end.isoString)
	})

	const borderClass = computed(() => (isLive.value ? 'border-emerald-600' : 'border-white'))
	const buttonClass = computed(() => (isLive.value ? 'bg-sky-500 hover:bg-sky-600' : 'bg-zinc-700'))

	const deleteMeeting = async () => {
		if (!props.meeting) return
		deleting.value = true
		try {
			await deleteMeetingApi(props.meeting.id)
			emit('refresh')
		} catch (error) {
			console.log('error deleting the meeting', error)
		} finally {
			deleting.value = false
		}
	}
</script>

<template>
	<div class="rounded-lg bg-zinc-800 p-4" :class="borderClass">
		<div class="grid gap-2">
			<div class="text-sm font-medium text-white">{{ meeting?.timeZone }}</div>
			<div class="grid gap-1">
				<p class="text-sm text-zinc-300">{{ displayDate(meeting?.startTime?.isoString) }}</p>
			</div>
		</div>
		<div class="mt-4 flex w-full items-center justify-between">
			<NuxtLink
				:to="`/meeting/${meeting?.id}`"
				class="flex gap-2 rounded-md px-3 py-2 text-sm font-medium leading-tight text-white shadow"
				:class="buttonClass"
				@click="video.joinMeeting"
			>
				Join
			</NuxtLink>
			<button
				v-if="isLeader && pod.id === meeting?.podRef?.id"
				class="rounded-md bg-red-600 px-3 py-2 text-sm font-medium leading-tight text-white shadow"
				@click="deleteMeeting"
			>
				Delete
				<span v-if="deleting">deleting...</span>
			</button>
		</div>
	</div>
</template>
