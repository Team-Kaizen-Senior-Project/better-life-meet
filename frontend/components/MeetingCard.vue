<script setup>
	const props = defineProps({
		meeting: Object,
		isLive: Boolean,
	})
	const video = useVideoStore()

	const borderClass = computed(() => (props.isLive ? 'border-emerald-600' : 'border-white'))
	const buttonClass = computed(() => (props.isLive ? 'bg-sky-500 hover:bg-sky-600' : 'bg-zinc-700'))
</script>

<template>
	<div class="rounded-lg bg-zinc-800 p-4" :class="borderClass">
		<div class="grid gap-2">
			<div class="text-sm font-medium text-white">{{ meeting.title }}</div>
			<div class="grid gap-1">
				<p class="text-sm text-zinc-300">{{ meeting.date }}</p>
				<p class="text-sm text-zinc-300">{{ meeting.time }}</p>
			</div>
		</div>
		<div class="flex w-full items-center justify-between">
			<NuxtLink
				:to="`/meeting/${meeting.id}`"
				class="mt-4 flex gap-2 rounded-md px-3 py-2 text-sm font-medium leading-tight text-white shadow"
				:class="buttonClass"
				@click="video.joinMeeting"
			>
				Join
			</NuxtLink>
		</div>
	</div>
</template>
