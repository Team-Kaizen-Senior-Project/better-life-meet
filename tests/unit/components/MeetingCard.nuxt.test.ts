import { describe, it } from 'vitest'
import { mount } from '@vue/test-utils'
import MeetingCard from '@/components/MeetingCard.vue'
import { nextTick } from 'vue'
import {mountSuspended} from "@nuxt/test-utils/runtime";

describe('MeetingCard', () => {
    it('MeetingCard displays correctly', async () => {
        //mocks
        const useDateMock = vi.fn().mockReturnValue({
            display: vi.fn(),
            displayHours: vi.fn(),
            dayjs: vi.fn(),
        })
        const useApiMock = vi.fn().mockReturnValue({
            deleteMeeting: vi.fn(),
            updateMeeting: vi.fn(),
            getVimeoVideo: vi.fn(),
        })
        const usePodStoreMock = vi.fn().mockReturnValue({
            state: { customerState: {} },
            isLeader: vi.fn(),
        })
        const useCustomerStoreMock = vi.fn().mockReturnValue({
            isAdmin: vi.fn(),
        })
        const useMediaStoreMock = vi.fn().mockReturnValue({})
        const useMeetingCountdownMock = vi.fn().mockReturnValue({
            renderText: vi.fn(),
        })

        const meetingMock = {
            id: '1',
            vimeoId: '123',
            startTime: { isoString: '2022-01-01T00:00:00Z' },
            endTime: { isoString: '2022-01-01T01:00:00Z' },
        }

        const wrapper = mount(MeetingCard, {
            props: {
                meeting: meetingMock,
                isFirst: true,
            },
            global: {
                mocks: {
                    useDate: useDateMock,
                    useApi: useApiMock,
                    usePodStore: usePodStoreMock,
                    useCustomerStore: useCustomerStoreMock,
                    useMediaStore: useMediaStoreMock,
                    useMeetingCountdown: useMeetingCountdownMock,
                },
            },
        })

        await nextTick()

        const podAccountabilityMeeting = wrapper.find('[data-testid="pod-accountability-meeting"]')
        expect(podAccountabilityMeeting.exists()).toBe(true)
    })
})
