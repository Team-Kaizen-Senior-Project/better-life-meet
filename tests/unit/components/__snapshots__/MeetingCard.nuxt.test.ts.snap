// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MeetingCard > MeetingCard displays correctly 1`] = `
"<div class="rounded-lg bg-zinc-800 p-4 border-white">
  <div class="grid gap-2">
    <div class="text-sm font-medium text-white">Pod accountability meeting</div>
    <div class="grid gap-2">
      <p class="text-sm text-zinc-300">Mar 29 2024</p>
      <p class="text-sm text-zinc-300"><span>9:43 AM - 9:43 AM</span></p>
      <!--v-if-->
    </div>
  </div>
  <div class="mt-4 flex w-full items-center justify-end">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;
